src/snyk/advisor/services/advisorService.ts

```diff
@@ -0,0 +1,128 @@
+import { Subscription } from 'rxjs';
+import { LineDecorations } from '../../common/editor/editorDecorator';
+import { ILog } from '../../common/logger/interfaces';
+import { ModuleParserProvider } from '../../common/services/moduleParserProvider';
+import { ImportedModule, Language } from '../../common/types';
+import { HoverAdapter } from '../../common/vscode/hover';
+import { IVSCodeLanguages } from '../../common/vscode/languages';
+import { IMarkdownStringAdapter } from '../../common/vscode/markdownString';
+import { getSupportedLanguage, isValidModuleName } from '../../common/vscode/parsing';
+import { IThemeColorAdapter } from '../../common/vscode/theme';
+import { Disposable, TextDocument, TextEditor } from '../../common/vscode/types';
+import { IVSCodeWindow } from '../../common/vscode/window';
+import { IVSCodeWorkspace } from '../../common/vscode/workspace';
+import { AdvisorScore } from '../advisorTypes';
+import EditorDecorator from '../editor/editorDecorator';
+import { IAdvisorApiClient } from './advisorApiClient';
+import AdvisorProvider from './advisorProvider';
+
+const SCORE_THRESHOLD = 0.7;
+export class AdvisorService implements Disposable {
+  protected disposables: Disposable[] = [];
+  protected advisorScanFinishedSubscription: Subscription;
+  protected activeEditor: TextEditor | undefined;
+
+  private readonly editorDecorator: EditorDecorator;
+
+  constructor(
+    private readonly window: IVSCodeWindow,
+    private readonly languages: IVSCodeLanguages,
+    private readonly advisorProvider: AdvisorProvider,
+    private readonly logger: ILog,
+    private readonly workspace: IVSCodeWorkspace,
+    private readonly advisorApiClient: IAdvisorApiClient,
+    private readonly themeColorAdapter: IThemeColorAdapter,
+    private readonly hoverAdapter: HoverAdapter,
+    private readonly markdownStringAdapter: IMarkdownStringAdapter,
+  ) {
+    this.editorDecorator = new EditorDecorator(
+      window,
+      this.languages,
+      this.themeColorAdapter,
+      this.advisorApiClient,
+      this.hoverAdapter,
+      this.markdownStringAdapter,
+    );
+  }
+
+  async activate(): Promise<boolean> {
+    this.activeEditor = this.window.getActiveTextEditor();
+    this.setEditorEvents();
+    if (!this.activeEditor) {
+      return false;
+    }
+
+    await this.handleEditorEvent(this.activeEditor.document);
+
+    return false;
+  }
+
+  setEditorEvents(): void {
```

Comment:
registerEditorListeners would be a better name. There are no events that we set here as per current function name.